packages/pancake-uikit/src/widgets/WalletModal/ConnectModal.tsx

```diff
@@ -1,40 +1,95 @@
-import React from "react";
-import styled from "styled-components";
-import { Link } from "../../components/Link";
-import { HelpIcon } from "../../components/Svg";
-import { Modal } from "../Modal";
-import WalletCard from "./WalletCard";
-import config from "./config";
-import { Login } from "./types";
+import React, { useState } from "react";
+import styled, { useTheme } from "styled-components";
+import Grid from "../../components/Box/Grid";
+import Box from "../../components/Box/Box";
+import getThemeValue from "../../util/getThemeValue";
+import Text from "../../components/Text/Text";
+import Heading from "../../components/Heading/Heading";
+import { Button } from "../../components/Button";
+import { ModalBody, ModalCloseButton, ModalContainer, ModalHeader, ModalTitle } from "../Modal";
+import WalletCard, { MoreWalletCard } from "./WalletCard";
+import config, { walletLocalStorageKey } from "./config";
+import { Config, Login } from "./types";
 
 interface Props {
   login: Login;
   onDismiss?: () => void;
+  displayCount?: number;
 }
 
-const HelpLink = styled(Link)`
-  display: flex;
-  align-self: center;
-  align-items: center;
-  margin-top: 24px;
+const WalletWrapper = styled(Box)`
+  border-bottom: 1px solid ${({ theme }) => theme.colors.cardBorder};
 `;
 
-const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => (
-  <Modal title="Connect to a wallet" onDismiss={onDismiss}>
-    {config.map((entry, index) => (
-      <WalletCard
-        key={entry.title}
-        login={login}
-        walletConfig={entry}
-        onDismiss={onDismiss}
-        mb={index < config.length - 1 ? "8px" : "0"}
-      />
-    ))}
-    <HelpLink href="https://docs.pancakeswap.finance/get-started/connection-guide" external>
-      <HelpIcon color="primary" mr="6px" />
-      Learn how to connect
-    </HelpLink>
-  </Modal>
-);
+/**
+ * Checks local storage if we have saved the last wallet the user connected with
+ * If we find something we put it at the top of the list
+ *
+ * @returns sorted config
+ */
+const getPreferredConfig = (walletConfig: Config[]) => {
```

Comment:
this logic should be in the app now, inside the user reducer (like all the config related to the user). The reducer is already serialized so we don't have to manipulate the local storage directly